src/users/interfaces/user.interface.ts

```diff
@@ -2,5 +2,13 @@ export interface User {
   id: number;
   name: string;
   login: string;
-  password: string;
-}
\ No newline at end of file
+  password?: string;
+}
+
+export interface UserWithoutPassword extends Omit<User, 'password'> { }
+
+export interface ValidatedUser {
+  id: number;
+  name: string;
+  login: string;
+}
```

Comment:
1. Se você já criou um `UserWithoutPassword` que remove a propriedade de password, ela ainda precisa ser opcional?
2. Se o ValidatedUser contem todos os dados de usuário menos a senha, ele não é igual o `UserWithoutPassword`? Pra que criar uma outra tipagem igual?

Reply:
Não estava nem sendo utilizado esse mano ai UserWithoutPassword, eu to achando que foi coisa que a gente colocou só pra vc me mostrar outras formas. Enfim, removi 🤓